lib/compilers/tic2000.js

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2019, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import path from 'path';
+
+import _ from 'underscore';
+
+import {BaseCompiler} from '../base-compiler';
+import {TiC2000AsmParser} from '../parsers/asm-parser-tic2000';
+
+export class TIC2000 extends BaseCompiler {
+    static get key() {
+        return 'tic2000';
+    }
+
+    constructor(info, env) {
+        super(info, env);
+        this.outputFilebase = this.compileFilename.split('.')[0];
+        this.asm = new TiC2000AsmParser(this.compilerProps);
+    }
+
+    optionsForFilter(filters, outputFilename) {
+        const options = ['-g', '-c', '-n', '--output_file=' + outputFilename];
+
+        filters.preProcessLines = _.bind(this.preProcessLines, this);
+
+        return options;
+    }
+
+    getOutputFilename(dirPath, outputFilebase) {
+        return path.join(dirPath, `${outputFilebase}.asm`);
+    }
+
+    exec(compiler, args, options_) {
+        const options = Object.assign({}, options_);
+        options.env = Object.assign({}, options.env);
+
+        if (this.compiler.includePath) {
+            options.env['INCLUDE'] = this.compiler.includePath;
+        }
+        if (this.compiler.libPath) {
+            options.env['LIB'] = this.compiler.libPath;
+        }
+        for (const [env, to] of this.compiler.envVars) {
+            options.env[env] = to;
+        }
+
+        options.env['TMP'] = process.env.TMP;
```

Comment:
In a "real" compiler explorer this is not safe: there's no tmp directory available to the compilers (they are jailed to write into a subdirectory of the tmp dir). I don't propose you change this; but in case we ever want to get it running on the main site, this (and indeed all the env stuff) needs to be changed.

Reply:
I'm unsure why I had these here in the first place. None of these are actually used by the compiler. I'll mark it as referencing a poor example many years ago. This function has been removed.